lib/hdcontroller.js

```diff
@@ -30,7 +32,20 @@ function _createRequest(req, log, callback) {
             return callback(error);
         }
         return callback(undefined, response);
-    }).on('error', callback);
+    }).on('error', (err) => {
+        // If callback was already called, log the error, but do not
+        // call the callback a second time.
+        if (callbackCalled) {
+            log.error('got socket error after response', {
```

Comment:
we will still get an error with this code:
* make the request
* get the response → set callbackCalled = true (line 23)
* 8 seconds later, timeout error → go in here and display an (unexpected) error

should the logic not be like this?
```
if (!callbackCalled) {
   callbackCalled = true;
   return callback(err);
}
if (err.code !== 'ERR_SOCKET_TIMEOUT') {
   log.error('got socket error after response', {err});
}
```

Reply:
Updated + squashed in https://github.com/scality/hdclient/pull/540/files#diff-da420d73d2cfe081498a59bce5b1cb59bd8d7614022658cb8bf66ef5858f4860R35